### ISSUE TITLE
[internal] Improve org parser -> use org-element.el

### ISSUE THREAD

**Issue by ardumont** [org-trello/org-trello#279]
I did not use all of org api for the parsing at first. In the mean time, org-mode clarified itself too.
There is a parsing org-element.el namespace. Use it.
Of course, clean up the useless tidbits hanging after that.

Should be especially useful for:
- org-trello's full compliance to org.
- keeping unknown org data from being lost during sync.

**Comment by iffyuva** [comment]
right now, do we use custom parser?

**Comment by ardumont** [comment]
> right now, do you use custom parser?

Partly, yes.
At the time when i booted org-trello, i did not know trello's api, org-mode's api (which changed a bit between 7 and 8) nor elisp's internals (not well).

It's still mostly built on org's api but not everything.
The real custom parser is for the checkbox which natively do not permit to hold metadata -> trello id, buffer checksum. This is built from scratch. 

So clearly, there are room for improvments.

**Comment by iffyuva** [comment]
yep, agreed. will look at org-element, and see if can offer some help.

**Comment by ardumont** [comment]
> yep, agreed. will look at org-element, and see if can offer some help.

It'd be great!
Thanks!

Cheers,

**Comment by ardumont** [comment]
For information, in 0.7.2, i've massively renamed all functions from `orgtrello-blabla/something` to orgtrello-blabla-something to respect fully elisp conventions...
I'm fighting with the ci right now but it will be soon merged.
